subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -988,3 +988,18 @@ The following _Java Object Serialization_ features are **not** supported:
 - the `readObjectNoData` method, if present, is never invoked;
 
 See link:{gradle-issues}13588[gradle/gradle#13588].
+
+[[config_cache:not_yet_implemented:storing_lambdas]]
+=== Objects Storing Lambdas
+
+Currently, Gradle cannot restore objects from the configuration cache that store user-provided lambdas.
```

Comment:
I'd like to mention that's not completely true because the configuration cache can indeed restore **serializable** user-provided lambdas and we make use of that fact in [our internal code](https://github.com/gradle/gradle/blob/ffdc459a127c358fccb41d2f1874e4d739a1924a/subprojects/base-services/src/main/java/org/gradle/api/internal/lambdas/SerializableLambdas.java#L26-L31).  Lambdas can be made _serializable_ by making the target `SAM` interface extend `java.io.Serializable`, for example.

I don't think we need to share that information here though, the final result might be more confusing and we already plan to solve the problem more generally before cc gets promoted to stable.

Reply:
👍 Thanks for the clarification.